src/skeleton/TLRAM.scala

```diff
@@ -0,0 +1,36 @@
+package sifive.skeleton
+
+import freechips.rocketchip.config.{Parameters}
+import freechips.rocketchip.diplomacy.{AddressSet, LazyModule}
+import freechips.rocketchip.tilelink.{TLRAM, TLFragmenter}
+import freechips.rocketchip.util.{ECCParams}
+
+case class TLRAMParams(
+  name: String,
+  address: AddressSet,
+  cacheable: Boolean = true,
+  executable: Boolean = true,
+  atomics: Boolean = false,
+  devName: Option[String] = None,
+  dtsCompat: Option[Seq[String]] = None
+)
+
+object TLRAMAttach {
```

Comment:
In the long term I'd prefer if this code lived in rocket-chip [here](https://github.com/chipsalliance/rocket-chip/blob/master/src/main/scala/tilelink/SRAM.scala). We can add this to the pile of technical debt that needs to be repaid when BAP/attachTo API are merged

Reply:
Yes I agree, we need to add a merged API to rocket-chip and start channeling everything through it. The SRAM stuff in particular would be a good use case to roll it out to address; we have a lot of roughly identical code to add "main memories"/LIMs/DLSs in various hierarchies to various buses.

My proposal would be to keep the BAP code as a legacy target here but try to figure out a way that `BlockDescriptorKey` can be translated in Config-space to become additional members of whatever the new device-adding key is, and instantiated via that mechanism instead of the trait mixed in here.

The one piece of functionality that BAP has that HasDevices does not is the ability to open up the testharness scope and add code there. But I think we could handle that through either the hierarchical location map or a separate `Field` that gives you access to that one special scope.